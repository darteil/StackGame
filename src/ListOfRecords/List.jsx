import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

const List = props => (
  <div className={styles.wrap}>
    <div className={styles['flex-block']}>
      <div className={styles.result}>
        <p>Best result</p>
        <p className={styles['result-stats']}>
          Count: <span>{props.topRecord}</span> Height stack: <span>{props.topHeight}</span>
        </p>
      </div>
      {
        props.listOfRecords.length > 0 &&
        <div className={styles.container}>
          <ul className={styles.list}>
            {
              props.listOfRecords.slice(0).reverse().map(record => (
                <li key={record.id}>
                  {record.time}
                  <div>
                    <span>count: {record.count}</span>
                    <span>height stack: {record.heightStack}</span>
                  </div>
                </li>
              ))
            }
          </ul>
        </div>
      }
    </div>
  </div>
);

List.propTypes = {
  listOfRecords: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    time: PropTypes.string,
    count: PropTypes.number,
    heightStack: PropTypes.number
  })),
  topRecord: PropTypes.number.isRequired,
  topHeight: PropTypes.number.isRequired
};

List.defaultProps = {
  listOfRecords: []
};

export default List;
