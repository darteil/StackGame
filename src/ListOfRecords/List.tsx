import React from 'react';
import { Record } from './types';
import styles from './styles.css';

interface IProps {
  listOfRecords: Record[];
  topRecord: number;
  topHeight: number;
}

const List = (props: IProps) => (
  <div className={styles.wrap}>
    <div className={styles['flex-block']}>
      <div className={styles.result}>
        <p>Best result</p>
        <p className={styles['result-stats']}>
          Count: <span>{props.topRecord}</span> Height stack: <span>{props.topHeight}</span>
        </p>
      </div>
      {props.listOfRecords.length > 0 && (
        <div className={styles.container}>
          <ul className={styles.list}>
            {props.listOfRecords
              .slice(0)
              .reverse()
              .map(record => (
                <li key={record.id}>
                  {record.time}
                  <div>
                    <span>count: {record.count}</span>
                    <span>height stack: {record.heightStack}</span>
                  </div>
                </li>
              ))}
          </ul>
        </div>
      )}
    </div>
  </div>
);

export default List;
