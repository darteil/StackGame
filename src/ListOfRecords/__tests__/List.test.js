import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import List from 'App/ListOfRecords/List';

describe('ListOfRecords component', () => {
  it('should render correctly', () => {
    const mockListOfRecords = [
      {
        id: 1,
        time: 'September 24th 2018, 16:15:26',
        count: 3,
        heightStack: 3
      },
      {
        id: 2,
        time: 'September 29th 2018, 16:11:32',
        count: 16,
        heightStack: 16
      }
    ];
    const initialState = {
      GameData: {
        listOfRecords: mockListOfRecords,
        topRecord: 16,
        topHeightStack: 16
      }
    };
    const output = shallow(
      <List
        listOfRecords={initialState.GameData.listOfRecords}
        topHeight={initialState.GameData.topHeightStack}
        topRecord={initialState.GameData.topRecord}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
