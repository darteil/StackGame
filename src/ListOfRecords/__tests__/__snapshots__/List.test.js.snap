// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfRecords component should render correctly 1`] = `
<div>
  <div>
    <div>
      <p>
        Best result
      </p>
      <p>
        Count: 
        <span>
          16
        </span>
         Height stack: 
        <span>
          16
        </span>
      </p>
    </div>
    <div>
      <ul>
        <li
          key="2"
        >
          September 29th 2018, 16:11:32
          <div>
            <span>
              count: 
              16
            </span>
            <span>
              height stack: 
              16
            </span>
          </div>
        </li>
        <li
          key="1"
        >
          September 24th 2018, 16:15:26
          <div>
            <span>
              count: 
              3
            </span>
            <span>
              height stack: 
              3
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
